sbol/object.py

```diff
@@ -283,14 +284,43 @@ def getProperties(self):
         owned_objects_uris = [p for p in self.owned_objects.keys()]
         return property_uris + owned_objects_uris
 
+    @deprecated(version='3.0.0', reason='Use extension properties instead')
     def setPropertyValue(self, property_uri, val):
         """Set and overwrite the value for a user-defined annotation property.
 
         :param property_uri:
-        :param val: Either a literal or URI value.
+        :param val: A string
         :return: None
         """
-        raise NotImplementedError("Not yet implemented")
+        if not isinstance(val, str):
+            raise TypeError('%r is not a string', val)
+        # Ensure that the property is a URIRef
+        property_uri = rdflib.URIRef(property_uri)
+        # If there is effectively no value ('', None, [], {}, etc.)
+        # clear out the value
+        if not val:
```

Comment:
The only way for line 301 to be true is `if val == ''` no?

Reply:
Yes, that is correct. The code got rearranged and the comment preceding line 301 did not. I'll fix up the comment. Thanks for pointing this out.